aptos-move/move-examples/token_objects/sources/token.move

```diff
@@ -68,25 +77,27 @@ module token_objects::token {
     ): ConstructorRef {
         let creator_address = signer::address_of(creator);
         let seed = create_token_seed(&collection, &name);
-        let creator_ref = object::create_named_object(creator, seed);
-        let object_signer = object::generate_signer(&creator_ref);
+        let constructor_ref = object::create_named_object(creator, seed);
+        let object_signer = object::generate_signer(&constructor_ref);
 
-        collection::increment_supply(&creator_address, &collection);
+        let id = collection::increment_supply(&creator_address, &collection);
         let token = Token {
             collection,
+            collection_id: option::get_with_default(&mut id, 0),
             creator: creator_address,
             description,
             mutability_config,
             name,
             creation_name: option::none(),
             uri,
+            mutation_events: object::new_event_handle(&object_signer),
         };
         move_to(&object_signer, token);
 
         if (option::is_some(&royalty)) {
-            collection::init_royalty(&object_signer, option::extract(&mut royalty))
+            royalty::init(&constructor_ref, option::extract(&mut royalty))
```

Comment:
This is beautiful! We finally got true modularity in Move - before it was just all friend functions and superficial connection between component modules. With objects and object signer (although arguably may be not the best yet), this is a much deeper connection between the modules, for example between token and royalty here. Royalty truly owns the Royalty resources and they're still grouped into the same core group that represents the Token object.

We should definitely highlight this design principle more as I strongly believe this is one strong benefit of Objects that will elevate the ecosystem to next level of designing smart contracts and applications.

Reply:
One interesting thought here though - what's the rationale for passing signer vs ExtendRef (or a reference of it) to royalty for adding resources?

Reply:
Because we only want to add it to an object and as part of extension.... otherwise we change the privilege model.